drivers/i2c/i2c_ll_stm32.c

```diff
@@ -18,6 +19,23 @@
 #define SYS_LOG_LEVEL CONFIG_SYS_LOG_I2C_LEVEL
 #include <logging/sys_log.h>
 
+#if defined(CONFIG_SOC_SERIES_STM32F3X) || defined(CONFIG_SOC_SERIES_STM32F0X)
+/*
+ * STM32F0/3 I2C's independent clock source supports only
+ * HSI and SYSCLK, not APB1. We force I2C clock source to SYSCLK.
+ * I2C2 on STM32F0 uses APB1 clock as I2C clock source
+ */
+#define LL_I2C_NO_CLOCKSOURCE 0x12345678 /* dummy value */
+#define LL_I2C_1_CLOCKSOURCE LL_RCC_I2C1_CLKSOURCE_SYSCLK
+#if defined(CONFIG_SOC_SERIES_STM32F3X)
+#define LL_I2C_2_CLOCKSOURCE LL_RCC_I2C2_CLKSOURCE_SYSCLK
+#else
+/* I2C2 on STM32F0 uses APB1 clock - can't set clock source */
+#define LL_I2C_2_CLOCKSOURCE LL_I2C_NO_CLOCKSOURCE
+#endif
+#define LL_I2C_3_CLOCKSOURCE LL_RCC_I2C3_CLKSOURCE_SYSCLK
+#endif
+
```

Comment:
Can't this go in a separate changeset ? Please stick to codegen only changes

Reply:
I had to do this change to use the codegen devicedeclare easily.

Reply:
Agree with @superna9999 , difficult to review and comment on mix of several new additions.
Please remove the clock source part and introduce it in another PR

Reply:
This PR is about introducing codegen for driver intiantiation. The PR includes various changes that I applied to make the drivers work with codegen (or better work). Currently it is a massive maintenance/ rebase effort that increases with every changeset to be managed. If codegen is accepted I can restructure the driver changes.